samples/subsys/fs/shell/README.rst

```diff
@@ -0,0 +1,152 @@
+.. _fs_shell-sample:
+
+File system shell example
+#########################
+
+Overview
+********
+
+This example provides shell access to a NFFS file system partition in flash.
+
+Requirements
+************
+
+A board with NFFS file system support and UART console
+
+Building
+********
+
+Native Posix
+============
+
+Before starting a build make, sure that the i386 pkconfig directory is in your
+search path.
+
+.. code-block:: console
+
+  export PKG_CONFIG_PATH=/usr/lib/i386-linux-gnu/pkgconfig
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/fs/shell
+   :board: native_posix
+   :goals: build
+   :compact:
+
+See :ref:`native_posix` on how to connect to the UART.
+
+Reel Board
+==========
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/fs/shell
+   :board: reel_board
+   :goals: build
+   :compact:
+
+Running
+*******
+
+Once the board has booted, you will be presented with a shell prompt.
+All file system related commands are available as sub-commands of fs.
+
+Begin by mounting the NFSS file system.
+
+.. code-block:: console
+
+  fs mount nfss /nffs
+
+Files System Shell Commands
+===========================
+
+Mount
+-----
+
+Mount a file system partition to a given mount point
+
+.. code-block:: console
+
+  fs mount (nffs|fat) <path>
+
+Ls
+--
+
+List all files and directories in a given path
```

Comment:
I have no experience with the filesystem subsystem, but I can't even get to list an empty mount point.
```
uart:~$ fs mount nffs /nffs
Successfully mounted fs:/nffs
uart:~$ fs ls /nffs
Unable to open /nffs (err -22)
```
or to mess with it from the FUSE mount side

Reply:
It is not you, I did hit the same issue.
After looking through the FS code it seems that it is limitation in Zephyrs FS implementation it seems that current implementation expects the root slash.

So ```fs ls /nffs/``` should do the trick, btw I did put a workaround in the Fuse code to handle this case.